lib/gitsh/cli.rb

```diff
@@ -82,5 +84,15 @@ def option_parser
         end
       end
     end
+
+    def ensure_executable_git
+      IO.popen(env.git_command).close
```

Comment:
It doesn't seem like it. `time bin/gitsh test` where test is an empty file takes roughly 0.25s on my system.